kolibri/core/discovery/models.py

```diff
@@ -37,4 +49,48 @@ def available(self):
             self.save()
             return True
         except NetworkClientError:
+            if self.dynamic:
+                self.delete()
```

Comment:
Dynamic `NetworkObjects` are automatically pruned when they're discovered to be unavailable.